src/components/Inventory/withdraw/flow/SelectPurposeForm.tsx

```diff
@@ -247,13 +247,16 @@ export default function SelectPurposeForm(props: SelectPurposeFormProps): JSX.El
       plantingSiteId: isOutplant ? localRecord.plantingSiteId : undefined,
       plotId: isOutplant ? localRecord.plotId : undefined,
       facilityId: Number(selectedNursery as string),
-      batchWithdrawals: batches
-        .filter((batch) => batch.facility_id.toString() === selectedNursery)
-        .map((batch) => ({
-          batchId: batch.id,
-          notReadyQuantityWithdrawn: isSingleOutplant ? 0 : isSingleBatch ? batch.notReadyQuantity : 0,
-          readyQuantityWithdrawn: isSingleOutplant ? withdrawnQuantity : isSingleBatch ? batch.readyQuantity : 0,
-        })),
+      batchWithdrawals:
+        isSingleBatch && !isSingleOutplant
+          ? localRecord.batchWithdrawals
+          : batches
+              .filter((batch) => batch.facility_id.toString() === selectedNursery)
+              .map((batch) => ({
+                batchId: batch.id,
+                notReadyQuantityWithdrawn: 0,
+                readyQuantityWithdrawn: isSingleOutplant ? withdrawnQuantity || 0 : 0,
+              })),
```

Comment:
use temporal values for setting quantities in single batch mode and use them here?